lib/browser/browser.ts

```diff
@@ -183,25 +207,52 @@ class OktaAuthBrowser extends OktaAuthBase implements OktaAuth, SignoutAPI {
 
     this.emitter = new Emitter();
     this.tokenManager = new TokenManager(this, args.tokenManager);
+    this.authStateManager = new AuthStateManager(this);
   }
 
   signIn(opts) {
-    opts = clone(opts || {});
-    const _postToTransaction = (options?) => {
-      delete opts.sendFingerprint;
-      return postToTransaction(this, '/api/v1/authn', opts, options);
+    const loginWithCredential = (opts) => {
+      opts = clone(opts || {});
+      const _postToTransaction = (options?) => {
+        delete opts.sendFingerprint;
+        return postToTransaction(this, '/api/v1/authn', opts, options);
+      };
+      if (!opts.sendFingerprint) {
+        return _postToTransaction();
+      }
+      return this.fingerprint()
+      .then(function(fingerprint) {
+        return _postToTransaction({
+          headers: {
+            'X-Device-Fingerprint': fingerprint
+          }
+        });
+      });
     };
-    if (!opts.sendFingerprint) {
-      return _postToTransaction();
-    }
-    return this.fingerprint()
-    .then(function(fingerprint) {
-      return _postToTransaction({
-        headers: {
-          'X-Device-Fingerprint': fingerprint
+
+    const loginWithRedirect = async (opts: SigninWithRedirectOptions = {}) => {
+      if(this._pending.handleLogin) { 
+        // Don't trigger second round
+        return;
+      }
+  
+      this._pending.handleLogin = true;
+      this.setFromUri(opts.fromUri);
```

Comment:
if `opts.fromUri` is undefined/null/empty string, do we want to set it? What if a user calls `setFromUri` with a valid URL, then they call `loginWithRedirect` with no options? Along with lifting the methods, are there any tests we should lift?

Reply:
seems like lifted tests were hidden by Github (need to click load diff to see it)
https://github.com/okta/okta-auth-js/pull/453/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md#diff-f7ba71daf0a700f1216bdfb283213838R550